searx/locales.py

```diff
@@ -90,74 +118,37 @@ def get_translations():
     return _flask_babel_get_translations()
 
 
-def get_locale_descr(locale, locale_name):
-    """Get locale name e.g. 'Français - fr' or 'Português (Brasil) - pt-BR'
+_TR_LOCALES: list[str] = []
 
-    :param locale: instance of :py:class:`Locale`
-    :param locale_name: name e.g. 'fr'  or 'pt_BR' (delimiter is *underscore*)
-    """
-
-    native_language, native_territory = _get_locale_descr(locale, locale_name)
-    english_language, english_territory = _get_locale_descr(locale, 'en')
-
-    if native_territory == english_territory:
-        english_territory = None
 
-    if not native_territory and not english_territory:
-        if native_language == english_language:
-            return native_language
-        return native_language + ' (' + english_language + ')'
+def get_translation_locales() -> list[str]:
```

Comment:
Why this function is not in `update_locales.py`, this is the only place where it is used.
Looking at `_TR_LOCALES`, I guess the intend is to use this function elsewhere at runtime? 
But `LOCALE_NAMES` already provides the same information, with the labels.

Reply:
> this is the only place where it is used.

My intention was to have most of the *business logic* in SearXNG / just hide the problematic `babel.Locale.parse` in update files.

> `LOCALE_NAMES` already provides the same information

The number of items in the list of `LOCALE_NAMES` is different from what `get_translation_locales()` returns / there are addition items from `ADDITIONAL_TRANSLATIONS` and `LOCALE_BEST_MATCH`.

If you insist, I can move this function to the update script / but for me it feels better to have it here in the `locale` module.

Reply:
I don't understand the use case. As far I remember / understand, we never had the need to get the result of `get_translation_locales()`.

The idea is to have everything in one place. That's why the initial commit moved and saved the values of `ADDITIONAL_TRANSLATIONS` and `LOCALE_BEST_MATCH` : to understand the code, everything is one file. As usual YMMV.

Reply:
> I don't understand the usage for `get_translation_locales()`, may I ask you to provide one?

Its documented in https://docs.searxng.org/src/searx.locales.html

![image](https://github.com/searxng/searxng/assets/554536/35f89c41-389b-4974-9337-7cfd756f0023)

From my POV the "how this list is generated?" is a part of our application model and should not been hidden somewhere in a update file / update tools should IMO just call some library methods but should not implement the application model partial.

I understand your point of view, but for me it just feels better if this list is generated by a function that is in the namespace `searx.locales` for the reasons mentioned.    

A newcomer will usually not look in an update script to see how a list of translations of the application is built.

> this is the only place where it is used.

It is also irrelevant to me whether this function is only used once / in the update script.

There are many functions that are only used once, but are therefore not in the namespace in which they are used. Now we have all the `locale` logic bundled in the `searx.locale` module .. where it is also [documented at a single point](https://docs.searxng.org/src/searx.locales.html).